### PR TITLE
Templates fixes

### PR DESCRIPTION
**Pull request by CharlieSYH**
## Info
- **Description**: 
Made a few edits to the templates to make them easier to read and edit.

## Before merging,
  - [x] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked related doc issue to this PR in **Development**. 
  - [ ] I have accquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [x] I have checked the doc site preview and the updated parts look all good. <details><summary>How?</summary>image</details>
  - [x] (For version-specific PR) I have confirmed that this PR is for a released version. If the software version is not released, put it on hold.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
+
+- **Related code PR**: 
+[ Provide link to the code PR here. For example, https://github.com/singularity-data/risingwave/pull/4085. Delete this part if there's no code PR related. ]
+
+- **Notes**: 
+[ Any additional information? ]
+
+<!--You DON'T need to edit the following sections when creating this pull request.-->
 
 ## Before merging,
```
Comment (hengm3467) [inline .github/pull_request_template.md]: Maybe remove the comma at the end?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
+☑️ if true or not applicable:
```
Comment (hengm3467) [inline .github/pull_request_template.md]: This statement is a bit confusing. Maybe remove it?

Comment (CharlieSYH) [reply .github/pull_request_template.md]: ok

File: .github/pull_request_template.md (hunk 1 above)
Comment (CharlieSYH) [reply .github/pull_request_template.md]: ok

**Review by hengm3467 — APPROVED** [review]
LGTM

**Pull request merged**